src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -240,4 +253,31 @@ def new_test_configuration(self, package_manager_name, package_manager_component
         configuration['config_env'] = Constants.TEST
         # perform desired modifications to configuration
         return configuration
+
+    def get_lifecycle_manager_component(self,vm_context):
+        """ fnding life cycle manager based on vm and returning component name add in the prod configuration"""
+        azure_package_manager_component = LifecycleManagerAzure
+        arc_package_manager_component = LifecycleManagerArc
+        if(vm_context == Constants.VMType.AZURE):
+            return azure_package_manager_component
+        elif (vm_context == Constants.VMType.ARC):
+            return arc_package_manager_component
+        
+        return azure_package_manager_component
+   
+    def get_vm_environment(self):
+        """ detects vm type; how to check this only when it is Auto Assessment operation??? """
+        try:
+            metadata = "True"
+            request = urlreq.Request(Constants.IMDS_END_POINT)
+            request.add_header('Metadata',metadata)
+            res = urlreq.urlopen(request,timeout=2)
+
+            if(res.getcode() == 200):
+                return Constants.VMType.AZURE
+            else:
+                return Constants.VMType.ARC
+        except:
+            return Constants.VMType.ARC
```

Comment:
must log what the failure was before returning so code behavior can be explained looking only at logs

Reply:
Actually... if there's no legitimate reason for this exception, type should be set to unknown, and execution should stop (with the exception rethrown).